flows/actions/modifiers/status.go

```diff
@@ -0,0 +1,83 @@
+package modifiers
+
+import (
+	"encoding/json"
+
+	"github.com/nyaruka/goflow/assets"
+	"github.com/nyaruka/goflow/envs"
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/events"
+	"github.com/nyaruka/goflow/utils"
+)
+
+func init() {
+	registerType(TypeStatus, readStatusModifier)
+}
+
+// TypeStatus is the type of our status modifier
+const TypeStatus string = "status"
+
+// StatusModifier modifies the status of a contact
+type StatusModifier struct {
+	baseModifier
+
+	Status flows.ContactStatus `json:"status" validate:"eq=active|eq=blocked|eq=stopped"`
+}
+
+// NewStatus creates a new status modifier
+func NewStatus(status flows.ContactStatus) *StatusModifier {
+	return &StatusModifier{
+		baseModifier: newBaseModifier(TypeStatus),
+		Status:       status,
+	}
+}
+
+// Apply applies this modification to the given contact
+func (m *StatusModifier) Apply(env envs.Environment, assets flows.SessionAssets, contact *flows.Contact, log flows.EventCallback) {
+
+	clearStatic := false
+	changed := false
+
+	if m.Status == flows.ContactStatusBlocked && !contact.Blocked() {
+		contact.SetBlocked(true)
```

Comment:
you should update contact to have `Status()` and `SetStatus()`.. that's how we want to expose it in the engine, and it'll make this method much simpler